### PR TITLE
Add DOIs for production archives on Zenodo

### PR DESCRIPTION
**Pull request by zaneselvans**
Created initial versions of raw inputs for the v0.4.0 release on the production Zenodo server, including:
* `censusdp1tract`
* `eia860`
* `eia861`
* `eia923`
* `epacems`
* `ferc1`
* `ferc714`

I've run a full ETL on all the data (including CEMS), as well as the data validation, and generation of the CEMS Parquet files, and it all seems to work.

### CODE REVIEW

File: test/conftest.py [hunk 1]
```diff
@@ -248,8 +250,12 @@ def pudl_engine(ferc1_engine, live_pudl_db, pudl_settings_fixture,
...
 
 @pytest.fixture(scope='session')  # noqa: C901
-def pudl_settings_fixture(request, tmpdir_factory,  # noqa: C901
-                          live_ferc1_db, live_pudl_db):  # noqa: C901
+def pudl_settings_fixture(  # noqa: C901
+    request,  # noqa: C901
+    tmpdir_factory,  # noqa: C901
+    live_ferc1_db,  # noqa: C901
+    live_pudl_db,  # noqa: C901
+):  # noqa: C901
```
Comment (cmgosnell) [inline test/conftest.py:258]: do these really need to be on every line? that seems excessive.

also we really should break up this function although that is clearly out of scope of this PR.

File: test/datastore_test.py [hunk 2]
```diff
@@ -13,28 +16,30 @@ def test_doi_to_url(self, pudl_datastore_fixture):
...
         assert pudl_datastore_fixture.doi_to_url(fake_doi) == \
-            f"https://sandbox.zenodo.org/api/deposit/depositions/{number}"
+            f"{pudl_datastore_fixture.api_root}/deposit/depositions/{number}"
+
+    def test_sandbox_vs_production_dois(self):
+        """Check that Sandbox/Production DOIs are formatted as expected."""
+        for doi in datastore.DOI["sandbox"].values():
+            assert re.match(r"^10\.5072/zenodo\.[0-9]{5,10}$", doi) is not None  # noqa: FS003
+        for doi in datastore.DOI["production"].values():
+            assert re.match(r"^10\.5281/zenodo\.[0-9]{5,10}$", doi) is not None  # noqa: FS003
```
Comment (cmgosnell) [inline test/datastore_test.py:26]: it seems like there would be less duplication here if there was an argument here like `ds_type` or something that was either "sandbox" or "production". Or even something like:
```
for doi in list(datastore.DOI["production"].values()) + list(datastore.DOI["production"].values()):
    assert re.match(r"^10\.5281/zenodo\.[0-9]{5,10}$", doi) is not None

**Review by cmgosnell — CHANGES_REQUESTED** [review]

File: test/datastore_test.py (hunk 2 above)
Comment (zaneselvans) [reply test/datastore_test.py:26]: Well, as it is now, there are two different patterns (reflecting the sandbox `10.5072` vs. production `10.5281` DOI patterns) and it tests all the production DOIs and all the sandbox DOIs whenever the test is run -- to make sure there hasn't been any confusion about which ones go where, cut-and-paste errors, etc. in the `datastore` module. I thought about doing some kind of `zip(("sandbox", "production"), ("5072", "5281"))` thing to iterate over but it seemed a little opaque.

File: test/conftest.py (hunk 1 above)
Comment (zaneselvans) [reply test/conftest.py:258]: Not sure. I was just copying the formatting of the function as I found it.

**Pull request merged**